pkg/cluster/mpi/statefulset.go

```diff
@@ -118,14 +118,29 @@ func (c statefulSetComponent) Reconcile(ctx *core.Context) (ctrl.Result, error)
 
 func (c statefulSetComponent) Finalize(ctx *core.Context) (ctrl.Result, bool, error) {
 	cr := objToMPICluster(ctx.Object)
+
 	pvcListOpts := []client.ListOption{
 		client.InNamespace(cr.Namespace),
 		client.MatchingLabels(meta.MatchLabels(cr)),
 	}
-
 	err := actions.DeleteStorage(ctx, pvcListOpts)
+	if err != nil {
+		return ctrl.Result{}, false, fmt.Errorf("cannot delete storage: %w", err)
+	}
+
+	sts := &appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      workerStatefulSetName(cr),
+			Namespace: cr.Namespace,
+			Labels:    meta.StandardLabelsWithComponent(cr, ComponentWorker, cr.Spec.Worker.Labels),
+		},
+	}
+	err = actions.DeleteIfExists(ctx, sts)
```

Comment:
Deleting is not synchronous, (because of finalizers), do you know if that is handled here?

Reply:
What keeps the status updater alive? The pod count? I guess my question is:

Lets say the StatefulSet is able to remove pods (to satisfy the status updater), but maybe not volumes (unlikely, but this is a pathological example). 

This would call delete, and Kubernetes my return "Ok! (but its pending deletion)". Will that be detected anywhere?

I'm not suggesting we block, but requeue if it is only pending and check that it got deleted later.